src/ResourceManager/Network/Commands.Network/Common/NetworkClient.cs

```diff
@@ -41,5 +54,177 @@ public NetworkClient(INetworkManagementClient NetworkManagementClient)
+            }
+            else
+            {
+                throw new Exception(string.Format("Get-AzureRmVpnClientPackage Operation Failed as no valid status code received in response!"));
+            }
+
+            string operationId = result.RequestId;
+            #endregion
+
+            #region 2. Wait for Async operation to succeed and then Get the content i.e. VPN Client package Url from locationResults
+            //Microsoft.WindowsAzure.Commands.Utilities.Common.TestMockSupport.Delay(60000);
+
+            // 2. Wait for Async operation to succeed           
+            var locationResultsUrl = new Uri(new Uri(baseUrl + (baseUrl.EndsWith("/") ? "" : "/")), "subscriptions/{subscriptionId}/providers/Microsoft.Network/" +
+                "locations/westus.validation/operationResults/{operationId}").ToString();
+            locationResultsUrl = locationResultsUrl.Replace("{operationId}", Uri.EscapeDataString(operationId));
+            locationResultsUrl = locationResultsUrl.Replace("{subscriptionId}", Uri.EscapeDataString(NetworkManagementClient.SubscriptionId));
+            locationResultsUrl += "?" + string.Join("&", string.Format("api-version={0}", Uri.EscapeDataString(apiVersion)));
+
+            DateTime startTime = DateTime.UtcNow;
+            DateTime giveUpAt = DateTime.UtcNow.AddMinutes(3);
+
+            // Send the Get locationResults request for operaitonId till either we get StatusCode 200 or it time outs (3 minutes in this case)
+            while (true)
+            {
+                HttpRequestMessage newHttpRequest = new HttpRequestMessage();
+                newHttpRequest.Method = new HttpMethod("GET");
+                newHttpRequest.RequestUri = new Uri(locationResultsUrl);
+
+                if (NetworkManagementClient.Credentials != null)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    await NetworkManagementClient.Credentials.ProcessHttpRequestAsync(newHttpRequest, cancellationToken).ConfigureAwait(false);
+                }
+
+                HttpResponseMessage newHttpResponse = await httpClient.SendAsync(newHttpRequest, cancellationToken).ConfigureAwait(false);
+
+                if ((int)newHttpResponse.StatusCode != 200)
+                {
+                    if (DateTime.UtcNow > giveUpAt)
+                    {
+                        if (!string.IsNullOrEmpty(newHttpResponse.Content.ReadAsStringAsync().Result))
+                        {
+                            result.Body = newHttpResponse.Content.ReadAsStringAsync().Result;
+                        }
+                        else
+                        {
+                            string newResponseContent = await newHttpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                            throw new Exception(string.Format("Get-AzureRmVpnClientPackage Operation returned an invalid status code '{0}' with Exception:{1} while retrieving " +
```

Comment:
I think you want to throw an exception whenever you are past the 'giveUpAt' time, so this exception throw belongs outside the else block, right?